exporter/redis.go

```diff
@@ -471,23 +471,39 @@ func (e *Exporter) scrapeRedisHost(scrapes chan<- scrapeResult, addr string, idx
 		if _, err := c.Do("SELECT", k.db); err != nil {
 			continue
 		}
-		if tempVal, err := c.Do("GET", k.key); err == nil && tempVal != nil {
-			if val, err := strconv.ParseFloat(fmt.Sprintf("%s", tempVal), 64); err == nil {
-				e.keyValues.WithLabelValues(addr, e.redis.Aliases[idx], "db"+k.db, k.key).Set(val)
+
+		obtainedKeys := []string{}
+		if strings.Contains(k.key, "*") {
```

Comment:
Interesting.

What do you think about doing the `KEYS` request every time, wildcard or not and if `k.key` matches a key in the DB then it'll pick it up anyway? 
Might simplify the code a little as the else block in line 485 isn't needed.

Thoughts?